[Job Fix]: Run FSAL_VFS tests instead of FSAL_GLUSTER

Manimaran-MM opened the pull request:
The test configuration has been updated to replace FSAL_GLUSTER with FSAL_VFS. This aligns with the current supported backends and simplifies the test matrix.

Log: https://jenkins-nfs-ganesha.apps.ocp.cloud.ci.centos.org/job/fsal-vfs/1/

Note: This PR is created based on the suggestion from @ffilz  https://github.com/nfs-ganesha/ci-tests/pull/102#issuecomment-3151318277

Manimaran-MM committed:
job fix: run FSAL_VFS tests instead of FSAL_GLUSTER

The test configuration has been updated to replace FSAL_GLUSTER
with FSAL_VFS. This aligns with the current supported backends and
simplifies the test matrix.

Signed-off-by: Manimaran M <manim@redhat.com>

psathyan commented on build_scripts/build-fsal/build-fsal_vfs.sh:
```diff
@@ -0,0 +1,74 @@
+    ;;
+    8s)
+        yum install -y ${BUILDREQUIRES}
+        yum install --enablerepo=powertools -y ${BUILDREQUIRES_EXTRA}
```
We don't nodes on `7 && 8s`. Please remove them.

We can avoid the switch but there would `10s` support in the near future.

Manimaran-MM replied:
Fixed

psathyan commented on build_scripts/build-fsal/build-fsal_vfs.sh:
```diff
@@ -0,0 +1,74 @@
+mkdir build
+cd build
+
+( cmake ../src -DCMAKE_BUILD_TYPE=Maintainer -DUSE_FSAL_VFS=ON -DUSE_FSAL_GLUSTER=OFF -DUSE_FSAL_CEPH=OFF -DUSE_FSAL_RGW=OFF -DUSE_FSAL_GPFS=OFF -DMONITORING=ON && make) || touch FAILED
```
I believe `-DUSE_MONITORING=ON` is the right one but it is also enabled by default.

Manimaran-MM replied:
Fixed

psathyan commented on build_scripts/build-fsal/build-fsal_vfs.sh:
```diff
@@ -0,0 +1,74 @@
+    echo "Change marked as WIP, not posting result to GerritHub."
+    touch WIP
+fi
+
```
I don't see a reason why we need these lines of code. It's not being used.

Manimaran-MM replied:
Fixed

Manimaran-MM committed:
fix(jjb): escape Groovy DSL braces to avoid JJB parsing errors

Escaped all { and } in the embedded Groovy pipeline DSL using double braces ({{, }}) to prevent Jenkins Job Builder (JJB) from misinterpreting them as Python format placeholders.
This resolves the unexpected '{' in field name error when running jenkins-jobs test or update.

Also moved the groovy script to the script folder for better readability

Signed-off-by: Manimaran M <manim@redhat.com>

kalebskeithley reviewed (approved):
LGTM
